### PR TITLE
task13

### PR DESCRIPTION
**Pull request by ab-fpmi**

### CODE REVIEW

File: witter/src/main/java/dao/jdbc/UserDAOJDBC.java [hunk 1]
```diff
@@ -39,6 +39,26 @@
         }
     }
 
+    @Override
+    public Optional<User> getUserByName(String username) {
+        String query = "select * from user where name = ?";
+        try (Connection conn = dataSource.getConnection()) {
```
Comment (smiakchilo) [inline witter/src/main/java/dao/jdbc/UserDAOJDBC.java:45]: Please avoid nested `try`s

File: witter/src/main/java/dao/jdbc/UserDAOJDBC.java [hunk 2]
```diff
@@ -68,6 +88,27 @@
         }
     }
 
+    @Override
+    public Optional<User> saveUser(User user) {
+        String query = "update user set name = ? and password = ? where id = ?";
```
Comment (smiakchilo) [inline witter/src/main/java/dao/jdbc/UserDAOJDBC.java:93]: Does it actually work with `and`? Never seen such) Usually it is `UPDATE smth SET param1='1', param2='2'`...

File: witter/src/main/java/servlets/users/SessionServlet.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            return;
+        }
+
+        String userIdStr = (String) session.getAttribute("userId");
+        if (userIdStr == null) {
+            sendNotFound(resp);
+            return;
+        }
+
+        Integer userId = Integer.parseInt(userIdStr);
```
Comment (smiakchilo) [inline witter/src/main/java/servlets/users/SessionServlet.java:47]: I'd rather check for parseability beforehead. `StringUtils.isNumeric()` would do

File: witter/src/main/java/servlets/users/SessionServlet.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        }
+
+        resp.setStatus(200);
+    }
+
+    private Optional<RegistrationForm> getRegistrationForm(HttpServletRequest req) {
+        try {
+            String body = IOUtils.toString(req.getReader());
+            return registrationFormJSONSerializer.fromJson(body);
+        } catch (Throwable e) {
```
Comment (smiakchilo) [inline witter/src/main/java/servlets/users/SessionServlet.java:96]: Please always use a specific exception telling what exactly can happen, `Throwable` is generally a bad pattern, and in our project is a particular meme ) Practically, you cannot do nothing if a `Throwable` other than an `Exception` thrown, If it is, say, `OutOfMemoryError`...what would you do at all?)

File: witter/src/main/java/dao/jdbc/UserDAOJDBC.java (hunk 2 above)
Comment (ab-fpmi) [reply witter/src/main/java/dao/jdbc/UserDAOJDBC.java:93]: Oh, that's a glitch! Sure it should be comma separated!

File: witter/src/main/java/dao/jdbc/UserDAOJDBC.java (hunk 1 above)
Comment (ab-fpmi) [reply witter/src/main/java/dao/jdbc/UserDAOJDBC.java:45]: I've thought about it but couldn't find passable solution, could you suggest some? My problem is I need to get connection from the pool and then I need to make prepared statement. And both resources need to be released if I understand it correctly.

**Comment by skateray17** [comment]
FYI: changed destination to hide some diff

File: witter/src/main/java/dao/jdbc/UserDAOJDBC.java (hunk 1 above)
Comment (smiakchilo) [reply witter/src/main/java/dao/jdbc/UserDAOJDBC.java:45]: Yes but this it not a problem `try(Cloeseable first = new Something(); Closeable second = factory.getSomethingElse()) {}` will do, and both things will be closed as they must

**Pull request merged**